File: typescript_nestjs/libs/core/src/shared/base/domain/AggregateRoot.ts
import { DomainEvent } from '../../bus/domain/DomainEvent';

export abstract class AggregateRoot {

    private _domainEvents: DomainEvent[];

    constructor() {
        this._domainEvents = [];
    }

    pullDomainEvents(): DomainEvent[] {
        const events = this._domainEvents;
        this._domainEvents = [];
        return events;
    }

    record(event: DomainEvent): void {
        this._domainEvents.push(event);
    }
}

File: python_django/core/shared/base/domain/aggregate_root.py
from typing import List

from core.shared.bus.domain.domain_event import DomainEvent


class AggregateRoot(object):
    _domainEvents: List[DomainEvent]

    def __init__(self):
        self._domainEvents = []

    def pull_domain_events(self) -> List[DomainEvent]:
        events = self._domainEvents
        self._domainEvents = []
        return events

    def record(self, domain_event: DomainEvent) -> None:
        self._domainEvents.append(domain_event)
